### PR TITLE
Fix get source HEAD requests

### PR DESCRIPTION
**Pull request by jasontedor**
Get source HEAD requests incorrectly return a content-length header of
0. This commit addresses this by removing the special handling for get
source HEAD requests, and just relying on the general mechanism that
exists for handling HEAD requests in the REST layer.

Relates #21125

### CODE REVIEW

**Comment by jasontedor** [comment]
I'll address the get action in a follow-up.

**Comment by jasontedor** [comment]
retest this please

**Comment by jasontedor** [comment]
retest this please

**Review by nik9000 — APPROVED** [review]

**Comment by jasontedor** [comment]
Thanks @nik9000!

**Comment by jasontedor** [comment]
> I'll address the get action in a follow-up.

The follow-up PR for get action is #23186.

File: core/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java [hunk 1]
```diff
@@ -36,21 +36,27 @@
...
+        controller.registerHandler(HEAD, "/{index}/{type}/{id}/_source", this);
     }
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         final GetRequest getRequest = new GetRequest(request.param("index"), request.param("type"), request.param("id"));
         getRequest.operationThreaded(true);
         getRequest.refresh(request.paramAsBoolean("refresh", getRequest.refresh()));
-        getRequest.routing(request.param("routing"));  // order is important, set it after routing, so it will set the routing
+        getRequest.routing(request.param("routing"));
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java]: the removed comment seemed quite important. Doesn't it apply anymore?

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java]: I don't think the comment makes any sense. What do you think it means?

Comment (javanna) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java]: it used to mean that things break if you set parent before routing. I wonder if that is still true though, probably in the past the parent setter was also setting the routing value.

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java]: I don't think it's true, that's why I don't think the comment makes sense. These are just boring field setters.

Comment (javanna) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java]: you are right, that comment doesn't apply anymore, the underlying request changed a a while ago to separate routing and parent. I updated some other rest actions that had that same comment for the same reason. https://github.com/elastic/elasticsearch/commit/578853f264e8ddae5c919b1b7ca35524cb292d62

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java]: Nice, thanks @javanna.

**Pull request merged**